components/main-chef-wrapper/main.go

```diff
@@ -42,6 +54,118 @@ func createDotChef() {
 	os.Mkdir(path, 0700)
 }
 
+func createRubyEnvUnix() {
+	InstallerDir := "/opt/chef-workstation"
+	home, err := os.UserHomeDir()
+	installationPath := path.Join(home, ".chef/ruby-env.json")
+	result, err := exists(installationPath)
+	if err != nil {
+		log.Fatalf(err.Error())
+	}
+	if platform_lib.OmnibusInstall() {
+		if result != true {
+			if createEnvJsonUnix(InstallerDir, installationPath) {
+				return
+			}
+		}
+		if result == true && platform_lib.MatchVersions() != true {
+			if createEnvJsonUnix(InstallerDir, installationPath) {
+				return
+			}
+		}
+	} else {
+		if result != true {
+			if createEnvJsonUnix(InstallerDir, installationPath) {
+				return
+			}
+		}
+		if result && platform_lib.MatchVersions() {
+			if createEnvJsonUnix(InstallerDir, installationPath) {
+				return
+			}
+		}
+	}
+
+	platform_lib.InitializeRubyMap()
+}
+
+func createRubyEnvWindows() {
+	InstallerDir := `C:\opscode\chef-workstation`
+	home, err := os.UserHomeDir()
```

Comment:
i think this func should return error when something goes wrong we are logging error but continuing work .

Reply:
Yeah, since we are running script with a check before every command, we just log the error, so that other command flow is not broken. and when env runs, the flow breaks. Even if something goes wrong in script which creates .json file another command should not be impacted.